src/stubs/sm_input.c

```diff
@@ -1,25 +1,61 @@
 #include "reactive_stubs_support.h"
 
 #include <alloca.h>
+#include <stdlib.h>
 
-#define AD_SIZE 2
-
-void SM_ENTRY(SM_NAME) __sm_handle_input(uint16_t conn_id,
+void SM_ENTRY(SM_NAME) __sm_handle_input(conn_index conn_id,
                                          const void* payload, size_t len)
 {
-    if (conn_id >= SM_NUM_INPUTS)
-        return;
+    // search for the connection.
+    // Unfortunately, this operation is O(n) with n = number of connections
```

Comment:
not sure, but why don't you return the connection_id when the user calls `sm_set_key` and the requiring the connection_id to be passed as  a capability later on (then you can allocate sequentially and only need to check the idx is within bounds like it was w the existing sm_keys array)?

Reply:
Implemented and tested 👍